integration/spark/src/main/common/java/io/openlineage/spark/agent/util/PlanUtils.java

```diff
@@ -183,4 +189,28 @@ public static Path getDirectoryPath(Path p, Configuration hadoopConf) {
       return p;
     }
   }
+
+  public static List<Path> findRDDPaths(List<RDD<?>> fileRdds) {
+    return fileRdds.stream()
+        .flatMap(
+            rdd -> {
+              if (rdd instanceof HadoopRDD) {
+                HadoopRDD hadoopRDD = (HadoopRDD) rdd;
+                Path[] inputPaths = FileInputFormat.getInputPaths(hadoopRDD.getJobConf());
+                Configuration hadoopConf = hadoopRDD.getConf();
```

Comment:
Will this approach to getting the Hadoop Conf suffer the same issues that Mike fixed for Azure Blob File System?

He had to pull the Hadoop Configuration from the sparksession

https://github.com/OpenLineage/OpenLineage/issues/459#issuecomment-1017127691

Reply:
The idea here was to extract existing code from `[AbstractRDDNodeVisitor.java](https://github.com/OpenLineage/OpenLineage/pull/698#diff-62c25157c46802c0def59a4b6a0a9a863f123f4a1afe602d7516983db2a654c3)` to common utils class, so that it can get reused. 

Follow-up PR contains removal part -> https://github.com/OpenLineage/OpenLineage/pull/645/files#diff-62c25157c46802c0def59a4b6a0a9a863f123f4a1afe602d7516983db2a654c3R36

So, if there are tests related to fix you mention, this should not suffer the same issue.